app/controllers/api/v1/projects_controller.rb

```diff
@@ -111,6 +180,30 @@ def set_project
       end
     end
 
+    # Update circuit data Related methods
+
+    # FIXME: remove this logic after fixing production data
```

Comment:
What is wrong with production data ?

Reply:
I actually it was in the old simulator controller also 
if only `current_user.projects.friendly.find_by(id: params[:id])` is used it does not allow collebrator access 
thats why `current_user.projects.friendly.find_by(id: params[:id]) || Project.friendly.find(params[:id])` is being used currently 

we can update it to 
```
def set_user_project
  @project = current_user.projects.friendly.find_by(id: params[:id])

  if @project.nil?
    @project = Project.friendly.find(params[:id])
    unless @project.collaborators.include?(current_user)
      render json: { error: 'You do not have access to this project' }, status: :unauthorized
      return
    end
  end
end
```